### DISCUSSION TITLE
GNV1B Conversion with “GraceL1b2Orbit”

### DISCUSSION THREAD

**Discussion by grkal** [groops-devs/groops#158] [Q&A] [answered]
Dear GROOPS Team,

The topic of this message concerns the conversion of the GNV1B data file of GRACE-FO/GRACE into GROOPS format using the “GraceL1b2Orbit” program.

In the “GraceL1b2Orbit” program’s description of the GROOPS documentation the following is mentioned:

[_When GNV1B is used, the orbit can be rotated from the terrestrial reference frame (TRF) transformed into the celestial reference frame (CRF) by specifying earthRotation._]

As an example, I will use the GRACE-C orbit data file from 2020-05-01 which I tried to convert into GROOPS format without rotation, i.e. keep the coordinates expressed in the TRF. However, after specifying as “< none >” the value of the “earthRotation” config element the following error message appears:

[_2020-05-01_00-00-00: orbit not in CRF. Must specify earthRotation._]

Judging by the aforementioned program’s documentation, I thought that it would be possible for the TRF coordinates to be kept after using the “GraceL1b2Orbit” program in case the user decides to do so. I would like to keep, if possible, the coordinates expressed in the TRF because they could be needed for plotting purposes vs simulated orbits expressed in the TRF using the “SimulateOrbit” program, etc..

Is there any way with which the TRF coordinates of the GNV1B file could be kept as it is with GROOPS or there is currently no way to perform this?

Attached you will find the “GNV1B2GroopsFormat.zip” file which includes the GNV1B file for the test date [GNV1B_2020-05-01_C_04.txt] and the “GNV1B2GroopsFormat.xml” file which is a simple script I created just containing the “GraceL1b2Orbit” program and the subsequent error message.

Thank you very much in advance for your time and looking forward to your feedback.

Best regards,
Grigorios Kalimeris
Chair of Astronomical and Physical Geodesy
Technical University of Munich
[GNV1B2GroopsFormat.zip](https://github.com/groops-devs/groops/files/10834930/GNV1B2GroopsFormat.zip)

**Comment by tmayerguerr** [comment, marked as answer]
Dear Grigorios,

The program [GraceL1b2Orbit](https://groops-devs.github.io/groops/html/GraceL1b2Orbit.html) contains the needless check
```
  if(coord_ref == 'E' && !earthRotation)
    throw(Exception(time.dateTimeStr()+": orbit not in CRF. Must specify earthRotation."));
```
that produces the error message. The lines could savely be removed.

You can rotate instrument files (including orbits) afterwards with [InstrumentRotate](https://groops-devs.github.io/groops/html/InstrumentRotate.html) together with [InstrumentEarthRotation](https://groops-devs.github.io/groops/html/InstrumentEarthRotation.html).

regards
Torsten

**Comment by grkal** [comment]
Dear Torsten,

Thank you very much for your detailed reply about the [GraceL1b2Orbit](https://groops-devs.github.io/groops/html/GraceL1b2Orbit.html) program.

Best regards,
Greg
